examples/depth.rs

```diff
@@ -0,0 +1,193 @@
+use wgpu::winit;
+use wgpu_glyph::{GlyphBrushBuilder, Scale, Section};
+
+fn main() -> Result<(), String> {
+    env_logger::init();
+
+    // Initialize GPU
+    let instance = wgpu::Instance::new();
+
+    let adapter = instance.get_adapter(&wgpu::AdapterDescriptor {
+        power_preference: wgpu::PowerPreference::HighPerformance,
+    });
+
+    let mut device = adapter.request_device(&wgpu::DeviceDescriptor {
+        extensions: wgpu::Extensions {
+            anisotropic_filtering: false,
+        },
+        limits: wgpu::Limits { max_bind_groups: 1 },
+    });
+
+    // Open window and create a surface
+    let mut events_loop = winit::EventsLoop::new();
+    let window = winit::WindowBuilder::new()
+        .with_resizable(false)
+        .build(&events_loop)
+        .unwrap();
+    let surface = instance.create_surface(&window);
+
+    // Prepare swap chain
+    let render_format = wgpu::TextureFormat::Bgra8UnormSrgb;
+
+    let mut size = window
+        .get_inner_size()
+        .unwrap()
+        .to_physical(window.get_hidpi_factor());
+
+    let mut swap_chain = device.create_swap_chain(
+        &surface,
+        &wgpu::SwapChainDescriptor {
+            usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+            format: render_format,
+            width: size.width.round() as u32,
+            height: size.height.round() as u32,
+            present_mode: wgpu::PresentMode::Vsync,
+        },
+    );
+
+    let mut depth_view = create_depth_view(&device, size);
+
+    // Prepare glyph_brush
+    let inconsolata: &[u8] = include_bytes!("Inconsolata-Regular.ttf");
+    let mut glyph_brush = GlyphBrushBuilder::using_font_bytes(inconsolata)
+        .depth_stencil_state(wgpu::DepthStencilStateDescriptor {
+            format: wgpu::TextureFormat::D32Float,
+            depth_write_enabled: true,
+            depth_compare: wgpu::CompareFunction::Greater,
+            stencil_front: wgpu::StencilStateFaceDescriptor::IGNORE,
+            stencil_back: wgpu::StencilStateFaceDescriptor::IGNORE,
+            stencil_read_mask: 0,
+            stencil_write_mask: 0,
+        })
```

Comment:
Should we allow this level of configuration or could this be simply hardcoded internally?

Reply:
I actually like taking the entire `DepthStencilStateDescriptor`, because I'm actually using a difference comparison function. I can imagine that users might want to change other aspects of the depth stencil state as well.